packages/side-nav/src/SideNav.tsx

```diff
@@ -35,11 +157,122 @@ interface SideNavProps {
  * @param props.className Class name that will be applied to the root-level element.
  * @param props.children Content that will be rendered inside the root-level element.
  */
-function SideNav({ className, children, ...rest }: SideNavProps) {
+function SideNav({
+  className,
+  currentPath = '',
+  children,
+  id: idProp,
+  ...rest
+}: SideNavProps) {
+  const { Provider: ContextProvider } = SideNavContext;
+  const [collapsed, setCollapsed] = useState(false);
+  const [hover, setHover] = useState(false);
+  const [focus, setFocus] = useState(false);
+  const { usingKeyboard } = useUsingKeyboardContext();
+  const navId = useMemo(() => idProp ?? navIdAllocator.generate(), [idProp]);
+  const [
+    portalContainer,
+    setPortalContainer,
+  ] = useState<HTMLUListElement | null>(null);
+
+  // We visually expand the navigation when a user focuses on an element within the navigation
+  // while navigating via keyboard.
+  const focusExpand = usingKeyboard && focus;
+
+  // Nav element should have a programmatically-determinable label
+  validateAriaLabelProps(rest, 'SideNav');
+
+  // Global event listener for toggling the navigation.
+  useEventListener(
+    'keypress',
+    e => {
+      const disabledTagNames = ['INPUT', 'TEXTAREA'] as const;
+
+      // Disable toggling the side navigation when a user is typing in an input.
+      // The typing for useEventListener doesn't seem to like using event.target,
+      // so we disable this here.
+      // @ts-expect-error
+      const shouldToggle = disabledTagNames.includes(e.target?.tagName);
+
+      if (e.keyCode === keyMap.BracketLeft && !shouldToggle) {
+        setCollapsed(curr => !curr);
+      }
+    },
+    {
+      options: {
+        passive: true,
+      },
+    },
+  );
+
   return (
-    <nav className={cx(navStyles, className)} aria-label="side-nav" {...rest}>
-      <ul className={ulStyleOverrides}>{children}</ul>
-    </nav>
+    <Transition
+      in={collapsed && !hover && !focusExpand}
+      timeout={collapseDuration}
+    >
+      {state => (
+        <ContextProvider
+          value={{
+            navId,
+            currentPath,
+            collapsed,
+            setCollapsed,
+            portalContainer: portalContainer,
```

Comment:
should just be able to write portalContainer here

Reply:
Whoops, you're right.